drivers/node/aws/aws.go

```diff
@@ -45,16 +52,86 @@ func (a *aws) Init(sched string) error {
 	a.config = config
 	svc := ec2.New(sess, config)
 	a.svc = svc
+	a.schedDriver, err = scheduler.Get(sched)
+	if err != nil {
+		return err
+	}
+	a.svcSsm = ssm.New(sess, aws_pkg.NewConfig().WithRegion(a.region))
+	a.session = sess
 	instances, err := a.getAllInstances()
 	if err != nil {
 		return err
 	}
 	a.instances = instances
+	nodes := a.schedDriver.GetNodes()
+	for _, n := range nodes {
+		if n.Type == node.TypeWorker {
+			if err := a.TestConnection(n, node.ConnectionOpts{
+				Timeout: 1 * time.Minute,
+			}); err != nil {
+				return &node.ErrFailedToTestConnection{
+					Node:  n,
+					Cause: err.Error(),
+				}
+			}
+		}
+	}
 	return nil
 }
 
 func (a *aws) TestConnection(n node.Node, options node.ConnectionOpts) error {
-	return nil
+	var err error
+	instanceID, err := a.getNodeIDByPrivAddr(n)
+	if err != nil {
+		return &node.ErrFailedToTestConnection{
+			Node:  n,
+			Cause: fmt.Sprintf("failed to get instance ID for connection due to: %v", err),
+		}
+	}
+	command := "uptime"
+	param := make(map[string][]*string)
+	param["commands"] = []*string{
+		aws_pkg.String(command),
+	}
+	sendCommandInput := &ssm.SendCommandInput{
+		Comment:      aws_pkg.String(command),
+		DocumentName: aws_pkg.String("AWS-RunShellScript"),
+		Parameters:   param,
+		InstanceIds: []*string{
+			aws_pkg.String(instanceID),
+		},
+	}
+	sendCommandOutput, err := a.svcSsm.SendCommand(sendCommandInput)
+	if err != nil {
+		return &node.ErrFailedToTestConnection{
+			Node:  n,
+			Cause: fmt.Sprintf("failed to send command to instance %s: %v", instanceID, err),
+		}
+	}
+	if sendCommandOutput.Command == nil || sendCommandOutput.Command.CommandId == nil {
+		return fmt.Errorf("No command returned after sending command")
+	}
+	listCmdsInput := &ssm.ListCommandInvocationsInput{
+		CommandId: sendCommandOutput.Command.CommandId,
+	}
+	var ticker = time.NewTicker(time.Millisecond * 500)
+	var timeout = time.NewTimer(options.Timeout).C
+loop:
+	for {
```

Comment:
Use `task.DoRetryWithTimeout(...)` to run code with timeout logic. See ssh package for examples.

Reply:
fixed